recirq/quantum_chess/quantum_board_test.py

```diff
@@ -1284,3 +1305,278 @@ def test_split_capture_with_failed_measurement_outcome(board):
+    b.do_move(
+        move.Move(
+            "f4",
+            "d6",
+            move_type=enums.MoveType.JUMP,
+            move_variant=enums.MoveVariant.CAPTURE,
+        )
+    )
+    b.do_move(
+        move.Move(
+            "h6",
+            "d6",
+            move_type=enums.MoveType.JUMP,
+            move_variant=enums.MoveVariant.CAPTURE,
+        )
+    )
+    assert b.is_classical()
+    assert b.circuit == cirq.Circuit()
+    assert b.ancilla_count == 0
+
+
+@pytest.mark.parametrize("board", ALL_CIRQ_BOARDS)
+def test_undo_to_start_after_measurement(board):
+    """Splits piece on f8 to d6 and h6. Piece on f4 then captures piece on d6. Then piece on h6 captures d6
+   Then does three undo moves to return to initial position."""
+    b = board(u.squares_to_bitboard(["f4", "f8"]))
+    b.reset_starting_states = True
+
+    initial_board = b.get_full_squares_bitboard()
+    b.do_move(
+        move.Move(
+            "f8",
+            "d6",
+            target2="h6",
+            move_type=enums.MoveType.SPLIT_JUMP,
+            move_variant=enums.MoveVariant.BASIC,
+        )
+    )
+    b.do_move(
+        move.Move(
+            "f4",
+            "d6",
+            move_type=enums.MoveType.JUMP,
+            move_variant=enums.MoveVariant.CAPTURE,
+        )
+    )
+    b.do_move(
+        move.Move(
+            "h6",
+            "d6",
+            move_type=enums.MoveType.JUMP,
+            move_variant=enums.MoveVariant.CAPTURE,
+        )
+    )
+    assert b.is_classical()
+    assert b.circuit == cirq.Circuit()
+
+    assert b.undo_last_move()
+    assert b.undo_last_move()
+    assert b.undo_last_move()
+
+    assert b.get_full_squares_bitboard() == initial_board
+
+
+@pytest.mark.parametrize("board", ALL_CIRQ_BOARDS)
+def test_quantum_capture_with_forced_measurement(board):
+    """Splits piece on b1 to a1 and c1. Piece on a1 then captures piece on a5."""
+    b = board(u.squares_to_bitboard(["b1", "a5"]))
+    b.reset_starting_states = True
+    b.do_move(
+        move.Move(
+            "b1",
+            "a1",
+            target2="c1",
+            move_type=enums.MoveType.SPLIT_JUMP,
+            move_variant=enums.MoveVariant.BASIC,
+        )
+    )
+    b.do_move(
+        move.Move(
+            "a1",
+            "a5",
+            move_type=enums.MoveType.JUMP,
+            move_variant=enums.MoveVariant.CAPTURE,
+        )
+    )
+    assert b.is_classical()
+    assert b.circuit == cirq.Circuit()
+    assert b.ancilla_count == 0
+
+
+@pytest.mark.parametrize("board", ALL_CIRQ_BOARDS)
+def test_consecutive_quantum_captures_with_successful_measurement_outcome(board):
+    """Splits piece on b2 to b8 and h2, then splits piece on e5 to d6 and f4.
+   Piece on b8 then captures piece on d6, with successful measurement. Then piece on d6 captures piece on f4,
+   also with a successful measurement."""
+    b = board(u.squares_to_bitboard(["b2", "e5"]))
+    b.reset_starting_states = True
+    b.do_move(
```

Comment:
IMHO the `b.perform_moves('b2^b8h2:SPLIT_JUMP:BASIC', 'e5^d6f4:SPLIT_JUMP:BASIC', ...)` format is more compact and readable.